packages/core/src/lib/auth0-synchronization-service.ts

```diff
@@ -0,0 +1,78 @@
+import { User, UserWrite } from "@dotkomonline/types"
+import { UserService } from "../modules/user/user-service"
+import { Auth0Repository } from "./auth0-repository"
+
+// Id token returned from Auth0. We don't want core to depend on next-auth, so we duplicate the type here.
+type JWTToken = {
+  email?: string | null
+  sub?: string | null
+  name?: string | null
+  givenName?: string | null
+  familyName?: string | null
+}
+
+/**
+ * Synchronize users in a local database user table with Auth0.
+ */
+export interface Auth0SynchronizationService {
+  /**
+   * Creates a new user in the local database using the provided JWT token.
+   */
+  createNewUser: (token: JWTToken) => Promise<User>
+
+  /**
+   * Synchronizes the provided user with data from Auth0. Only updates user if the user's updatedAt field is more than 1 day ago.
+   */
+  handleSyncUserWithAuth0: (user: User) => Promise<User | undefined>
```

Comment:
:nerd_face: :point_up: 
```suggestion
  createUser: (token: JWTToken) => Promise<User>

  /**
   * Synchronizes the provided user with data from Auth0. Only updates user if the user's updatedAt field is more than 1 day ago.
   */
  synchronizeUser: (user: User) => Promise<User | undefined>
```